include more descriptive language for additional versions (#11260)
dotnet/docs: docs/csharp/language-reference/index.md

  
 [C# Language Specification](../../../_csharplang/spec/introduction.md)  
 The C# 6.0 language specification. This is a draft proposal for the C# 6.0 language. Version 5.0 has been released in December 2017 as the [Standard ECMA-334 5th Edition](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-334.pdf) document.

The features that have been implemented in C# versions after 6.0 are represented in language specification proposals. These documents describe the deltas to the language spec in order to add these new features. 

 [C# 7.0 Language Proposals](../../../_csharplang/proposals/csharp-7.0/pattern-matching.md)  
 There are a number of new features implemented in C# 7.0. They include pattern matching, local functions, out variable declarations, throw expressions, binary literals, and digit separators. This folder contains the specifications for each of those features.
  
 [C# 7.1 Language Proposals](../../../_csharplang/proposals/csharp-7.1/async-main.md)  

>  There are new features added in C# 7.1. First, you can write a `Task` or `Task<int>` returning `Main` method. This enables you to add the `async` modifier to `Main`. The `default` expression can be used without a type in locations where the type can be inferred. Also, tuple member names can be inferred. Finally, pattern matching can be used with generics.

Reviewer: ```suggestion
 There are new features added in C# 7.1. First, you can write a `Main` method that returns `Task` or `Task<int>`. This enables you to add the `async` modifier to `Main`. The `default` expression can be used without a type in locations where the type can be inferred. Also, tuple member names can be inferred. Finally, pattern matching can be used with generics.
```

 [C# 7.2 Language Proposals](../../../_csharplang/proposals/csharp-7.2/readonly-ref.md)  

>  C# 7.2 added a number of small features. You can pass arguments by readonly reference using the `in` keyword. There are a number of low-level changes to support compile time safety for `Span` and related types. You can use named arguments where later arguments are positional, in some situations. The `private protected` access modifier enables you to specify that callers are limited to derived types implemented in the same assembly. The `?:` operator can resolve to a reference to a variable. You can also format hexadecimal and binary numbers using a leading digit separator.   

Reviewer: ```suggestion
 C# 7.2 added a number of small features. You can pass arguments by readonly reference using the `in` keyword. There are a number of low-level changes to support compile-time safety for `Span` and related types. You can use named arguments where later arguments are positional, in some situations. The `private protected` access modifier enables you to specify that callers are limited to derived types implemented in the same assembly. The `?:` operator can resolve to a reference to a variable. You can also format hexadecimal and binary numbers using a leading digit separator.   
```

 [C# 7.3 Language Proposals](../../../_csharplang/proposals/csharp-7.3/blittable.md)  

>  C# 7.3 is another point release that includes several small updates. You can use new constraints on generic type parameters. Other changes make it easier to work with `fixed` fields, including using `stackalloc` allocations. Local variables declared with the `ref` keyword may be reasssigned to refer to new storage. You can place attributes on auto-implemented properties that target the compiler generated backing field. Expression variables can be used in initializers. Tuples can be compared for equality (or inequality). There have also been some improvements to overload resolution.

Reviewer: ```suggestion
 C# 7.3 is another point release that includes several small updates. You can use new constraints on generic type parameters. Other changes make it easier to work with `fixed` fields, including using `stackalloc` allocations. Local variables declared with the `ref` keyword may be reasssigned to refer to new storage. You can place attributes on auto-implemented properties that target the compiler-generated backing field. Expression variables can be used in initializers. Tuples can be compared for equality (or inequality). There have also been some improvements to overload resolution.
```
Perhaps explain what stackalloc allocations are, or link to stackalloc?

>  C# 8.0 is available in preview. The following proposals are the current versions of the specifications for those features. Some are more complete; some are still a work in progress.

Reviewer: Two separate paragraphs here seems a bit odd, since the others follow the format of a link followed by descriptive text in a separate paragraph. Perhaps consider combining them, or have the link first followed by two paragraphs?
